docs(upgrade): fix `upgrade` docs examples for v13

gkalpak opened the pull request:
_Builds on top of #43880._

##
This PR primarily fixes the SystemJS docs examples to be compatible with Angular v13 and re-enables the associated tests on CI.

See the individual commits for details.

mary-poppins commented:
You can preview 19e1fe4 at https://pr43868-19e1fe4.ngbuilds.io/.

mary-poppins commented:
You can preview 4c26fdc at https://pr43868-4c26fdc.ngbuilds.io/.

petebacondarwin commented on aio/content/guide/upgrade-setup.md:
````diff
@@ -123,6 +92,125 @@ Open a terminal window in the project folder and enter the following commands fo
+   ```
+
+5. In order to be able to load the latest Angular framework packages (in ES2015 format) correctly, replace the relevant entries in `src/systemjs.config.js`:
+    ```js
+    System.config({
+      // ...
+      map: {
+        // ...
+        '@angular/core': 'npm:@angular/core/fesm2015/core.mjs',
+        '@angular/common': 'npm:@angular/common/fesm2015/common.mjs',
+        '@angular/compiler': 'npm:@angular/compiler/fesm2015/compiler.mjs',
+        '@angular/platform-browser': 'npm:@angular/platform-browser/fesm2015/platform-browser.mjs',
+        '@angular/platform-browser-dynamic': 'npm:@angular/platform-browser-dynamic/fesm2015/platform-browser-dynamic.mjs',
+        '@angular/router': 'npm:@angular/router/fesm2015/router.mjs',
+        '@angular/forms': 'npm:@angular/forms/fesm2015/forms.mjs',
+        // ...
+      },
+      // ...
+    });
+    ```
````
Could this not be externalized, given that we have a load of `systemjs.config.js` files in our examples?

petebacondarwin replied:
And similarly for the snippets below...

gkalpak replied:
Good point! I forgot to do that. Fixed now.

petebacondarwin reviewed (approved):
One suggestion about the example snippets but happy to approve the rest.

gkalpak committed:
docs(upgrade): minor fixes to the `upgrade` guide

Includes the following fixes:
- Remove unused `format` attribute from `<code-example>` elements
- Explicitly mention that the starting point for the "Phonecat Upgrade
  Tutorial" is the 1.5.x version of the original Phonecat tutorial
  (since the original tutorial has since been updated to use AngularJS
  v1.8.x and use npm instead of bower).

gkalpak committed:
build(docs-infra): fix SystemJS-based docs examples

Previously, the SystemJS-based docs examples (typically `ngUpgrade`
examples) were not compatible with Angular v13 (due to the removal of
UMD bundles from the published packages).

This commit makes the examples compatible with Angular v13 and
re-enables the tests for all SystemJS-based examples on CI.

gkalpak committed:
build(docs-infra): include `systemjs` as a dependency to SystemJS-based docs examples

Due to an oversight, the `systemjs` package was previously not included
as a dependency in the boilerplate `package.json` file for
SystemJS-based docs examples, despite it being required by the app.

gkalpak committed:
fixup! build(docs-infra): fix SystemJS-based docs examples

mary-poppins commented:
You can preview 1e8add6 at https://pr43868-1e8add6.ngbuilds.io/.

atscott commented:
This PR was merged into the repository by commit 37e02750a707eaaa859e87c0c55a39a1c5361ec1.
